src/Runner.Worker/ActionCommandManager.cs

```diff
@@ -223,6 +223,24 @@ public void ProcessCommand(IExecutionContext context, string line, ActionCommand
                 throw new Exception("Required field 'name' is missing in ##[set-env] command.");
             }
 
+
+            foreach (var blocked in _setEnvBlockList)
+            {
+                if (string.Equals(blocked, envName, StringComparison.OrdinalIgnoreCase))
+                {
+                    // Log Telemetry and let user know they shouldn't do this
+                    var issue = new Issue()
+                    {
+                        Type = IssueType.Error,
+                        Message = $"Can't update {blocked} environment variable using ::set-env:: command."
+                    };
+                    issue.Data[Constants.Runner.InternalTelemetryIssueDataKey] = $"{Constants.Runner.UnsupportedCommand}_{envName}";
+                    context.AddIssue(issue);
+
+                    return;
```

Comment:
Minor: this `return;` statement was surprising to me to read at first.

I'm guessing that if we ever add more items to `_setEnvBlockList`, we want the runner to report an error on the first offending env var, instead of all of them, @TingluoHuang?

Reply:
I was just want to make sure we don't run 
```
context.Global.EnvironmentVariables[envName] = command.Data;
context.SetEnvContext(envName, command.Data);
context.Debug($"{envName}='{command.Data}'");
```